File: client/packages/api-sdk/src/apis/menu-api.ts
import { getClient } from '../config';
import type { CurrentUserMenuVo, MenuVo } from '../types/menu';

export const menuApi = {
  /** 获取当前用户菜单树+权限列表（GET /menus/current-user，需认证） */
  queryCurrentUserMenu(): Promise<CurrentUserMenuVo> {
    return getClient().request('/api/v1/menus/current-user');
  },

  /** 获取菜单树（需 iam:menu:list 权限） */
  tree(): Promise<MenuVo[]> {
    return getClient().request('/api/v1/menus');
  },

  /** 获取单个菜单详情 */
  getById(id: number): Promise<MenuVo> {
    return getClient().request(`/api/v1/menus/${id}`);
  },
};

File: server/mb-platform/platform-iam/src/main/java/com/metabuild/platform/iam/api/MenuApi.java
package com.metabuild.platform.iam.api;

import com.metabuild.platform.iam.api.vo.MenuVo;
import java.util.List;

/**
 * 菜单模块对外 API 接口（供跨模块调用）。
 */
public interface MenuApi {

    MenuVo getById(Long id);

    List<MenuVo> tree();

    /** 查询指定角色的菜单列表 */
    List<MenuVo> listByRoleId(Long roleId);
}
